docs/glossary.md

```diff
@@ -0,0 +1,152 @@
+### Data dictionary
+:   A JSON file that describes the information contained in columns from a tabular data file, 
+    along with the meaning and properties (format of numerical data, unique “levels” 
+    of categorical data, etc.) of values in each column. In the context of Neurobagel, 
+    the meanings of columns and column values are encoded using terms from standardised vocabularies.
+
+### Data model
+**Used interchangeably with**: data schema
```

Comment:
🍒 : I think visually, italics for "Used interchangeably with:" (e.g., _Used interchangeably with_: data schema) looks a little cleaner, since some of the definitions include bolded text as well.

This is subjective though, so will leave the style decision up to you :)